client/app/components/file_picker.coffee

```diff
@@ -225,27 +227,26 @@ FileItem = React.createClass
         li className: "file-item", key: @props.key,
             i className: "mime #{type} fa #{iconClass}"
             a
-                className: 'file-name',
-                target: '_blank',
-                onClick: @doDisplay,
-                href: file.url
-                'data-file-url': file.url
-                'data-file-name': file.generatedFileName
-                'data-file-type': file.contentType
+                'className'         : 'file-name'
+                'onClick'           : @doDisplay
+                'data-file-url'     : @state.tmpFileURL
+                'data-file-name'    : file.generatedFileName
+                'data-file-type'    : file.contentType
                 file.generatedFileName
 
             span className: 'file-size',
                 "\(#{(file.length / 1000).toFixed(1)}Ko\)"
             span className: 'file-actions',
                 a
-                    className: "fa fa-download"
-                    href: "#{file.url}?download=1"
+                    'className' : 'fa fa-download'
+                    'download'  : file.generatedFileName
+                    'href'      : getFileURL file
```

Comment:
@aenario : it must be the real `file.url`here otherwise feature "download" doesn't work. See my rollback commit: https://github.com/misstick/cozy-emails/commit/b6378cf11ea2ed38160268dbac9e50a280ad5c10#diff-4ef6be8edecdebfd711979b18edc3434L232